python/desc/twinkles/sprinkler.py

```diff
@@ -185,255 +184,317 @@ def __init__(self, catsim_cat, visit_mjd, specFileMap, sed_path,
-
-
-                        updated_catalog = np.append(updated_catalog, lensrow)
-
-                    #Now manipulate original entry to be the lens galaxy with desired properties
-                    #Start by deleting Disk and AGN properties
-                    if not np.isnan(row[self.defs_dict['galaxyDisk_magNorm']]):
-                        row[self.defs_dict['galaxyDisk_majorAxis']] = 0.0
-                        row[self.defs_dict['galaxyDisk_minorAxis']] = 0.0
-                        row[self.defs_dict['galaxyDisk_positionAngle']] = 0.0
-                        row[self.defs_dict['galaxyDisk_internalAv']] = 0.0
-                        row[self.defs_dict['galaxyDisk_magNorm']] = 999. #np.nan To be fixed post run1.1
-                        row[self.defs_dict['galaxyDisk_sedFilename']] = None
-                    row[self.defs_dict['galaxyAgn_magNorm']] = None #np.nan To be fixed post run1.1
-                    row[self.defs_dict['galaxyDisk_magNorm']] = 999. # To be fixed in run1.1
-                    row[self.defs_dict['galaxyAgn_sedFilename']] = None
-                    #Now insert desired Bulge properties
-                    row[self.defs_dict['galaxyBulge_sedFilename']] = defaultSpecMap[newlens['lens_sed']]
-                    row[self.defs_dict['galaxyBulge_redshift']] = newlens['ZLENS']
-                    row[self.defs_dict['galaxyDisk_redshift']] = newlens['ZLENS']
-                    row[self.defs_dict['galaxyAgn_redshift']] = newlens['ZLENS']
-                    row_lens_sed = Sed()
-                    row_lens_sed.readSED_flambda(str(self.galDir + newlens['lens_sed']))
-                    row_lens_sed.redshiftSED(newlens['ZLENS'], dimming=True)
-                    row[self.defs_dict['galaxyBulge_magNorm']] = matchBase().calcMagNorm([newlens['APMAG_I']], row_lens_sed,
-                                                                         self.bandpassDict) #Changed from i band to imsimband
-                    row[self.defs_dict['galaxyBulge_majorAxis']] = radiansFromArcsec(newlens['REFF'] / np.sqrt(1 - newlens['ELLIP']))
-                    row[self.defs_dict['galaxyBulge_minorAxis']] = radiansFromArcsec(newlens['REFF'] * np.sqrt(1 - newlens['ELLIP']))
-                    #Convert orientation angle to west of north from east of north by *-1.0 and convert to radians
-                    row[self.defs_dict['galaxyBulge_positionAngle']] = newlens['PHIE']*(-1.0)*np.pi/180.0
-
-                    if self.logging_is_sprinkled:
-                        row[self.defs_dict['galaxyAgn_is_sprinkled']] = 1
-                        row[self.defs_dict['galaxyBulge_is_sprinkled']] = 1
-                        row[self.defs_dict['galaxyDisk_is_sprinkled']] = 1
-
-                    #Replace original entry with new entry
-                    updated_catalog[rowNum] = row
+                rng = np.random.RandomState(galtileid % (2^32 -1))
```

Comment:
Should we change these 2^32 values with 2^64 values due to the integer limit change?

Reply:
What are we trying to achieve here?  Is the idea that each AGN should have its own unique random seed?

`np.random.RandomState` requires the seed to be a 32 bit integer.  `2**32=4.3*10**9`, which is less than the maximum `galaxy_id`, so we do need to cast this into a 32 bit integer.  That being said, `^` is a logical operator.  Should this line be `rng=np.random.RandomState(galtileid % (2**32-1))`?

Reply:
Now that I think about it yeah, we just wanted unique seeds but also a way to back everything out if we needed to. So, I think yes, just  `rng=np.random.RandomState(galtileid % (2**32-1))` is what we want.